cmd/opm/index/deprecate.go

```diff
@@ -0,0 +1,132 @@
+package index
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+	"github.com/operator-framework/operator-registry/pkg/lib/indexer"
+)
+
+var deprecateLong = templates.LongDesc(`
+	Deprecate operator bundles from an index.
+	
+	Deprecated bundles will no longer be installable. Bundles that are replaced by deprecated bundles will be removed enirely from the index.
+	
+	For example:
+
+		Given the update graph in quay.io/my/index:v1
+		1.4.0 -- replaces -> 1.3.0 -- replaces -> 1.2.0 -- replaces -> 1.1.0
+
+		Applying the command:
+		opm index deprecate --bundles "quay.io/my/bundle:1.3.0" --from-index "quay.io/my/index:v1" --tag "quay.io/my/index:v2"
+
+		Produces the following update graph in quay.io/my/index:v2
+		1.4.0 -- replaces -> 1.3.0 [deprecated]
+		
+	Deprecating a bundle that removes the default channel is not allowed. Changing the default channel prior to deprecation is possible by publishing a new bundle to the index.
```

Comment:
> Changing the default channel prior to deprecation is possible by publishing a new bundle to the index.
it shouldn't block this PR, but it's been brought up at the past 2 working groups that we need a way to change this without adding a bundle.